lib/widgets/tag_tiles.dart

```diff
@@ -0,0 +1,109 @@
+import 'package:flutter/material.dart';
+import '/theme/theme.dart';
+
+class TagTile extends StatefulWidget {
+  final String tagName;
+  final bool isChecked;
+
+  TagTile({required this.tagName, this.isChecked = false});
+
+  @override
+  _TagTileState createState() => _TagTileState();
+}
+
+class _TagTileState extends State<TagTile> {
+  late bool _isChecked;
+  late TextEditingController _controller;
+  bool _isEditing = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _isChecked = widget.isChecked;
+    _controller = TextEditingController(text: widget.tagName);
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  void _toggleCheckbox(bool value) {
+    setState(() {
+      _isChecked = value;
+    });
+  }
+
+  void _toggleEditMode() {
+    setState(() {
+      _isEditing = !_isEditing;
+    });
+  }
+
+  void _onSubmitted(String value) {
+    setState(() {
+      _isEditing = false;
+      _controller.text;
```

Comment:
line is redundant; i think you mean to assign `_originalText = value`

Reply:
done